internal/templates/render/render.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package render
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/abcxyz/pkg/cli"
+)
+
+type Command struct {
+	cli.BaseCommand
+
+	template        string
+	spec            string
+	dest            string
+	gitProtocol     string
+	logLevel        string
+	allowNonGitDest bool
+	forceOverwrite  bool
+	keepTempDirs    bool
+	inputs          map[string]string
+}
+
+// Desc implements cli.Command.
+func (c *Command) Desc() string {
+	return "Instantiate a template to setup a new app or add config files"
+}
+
+func (c *Command) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+  The {{ COMMAND }} command renders the given template into the given dest
```

Comment:
[nit ] Terminals has a default line wrap, when a line exceeds the width of the terminal window, the content automatically wraps to the next line. Instead of multiline strings with a fixed width, shall we make this one line here and delegate the wrapping behavior to terminal? Similar to the flag descs below.

Reply:
Most CLIs stop at ~80 characters. It automatically wraps: https://github.com/abcxyz/pkg/blob/main/cli/flags.go#LL690C6-L690C29

Reply:
Removed the manual line wrap.